packages/lib/CalendarService.ts

```diff
@@ -190,20 +190,44 @@ export default abstract class BaseCalendarService implements Calendar {
 
       const eventsToUpdate = events.filter((e) => e.uid === uid);
       return Promise.all(
-        eventsToUpdate.map((e) => {
+        eventsToUpdate.map((eventItem) => {
+          event = eventItem;
           return updateCalendarObject({
             calendarObject: {
-              url: e.url,
-              data: iCalString,
-              etag: e?.etag,
+              url: event.url,
+              data: iCalString?.replace(/METHOD:[^\r\n]+\r\n/g, ""),
+              etag: event?.etag,
             },
             headers: this.headers,
           });
         })
-      ).then((p) => p.map((r) => r.json() as unknown as NewCalendarEventType));
+      ).then((responses) =>
+        responses.map((response) => {
+          if (response.status > 199 && response.status < 207) {
+            const ret = {
+              uid,
+              type: this.credentials.type,
+              id: typeof event.uid === "string" ? event.uid : "-1",
+              password: "",
+              url: event.url,
+              additionalInfo: event.additionalInformation,
+            } as NewCalendarEventType;
```

Comment:
Since some providers return updated data while the others choose not to (sending 204 instead), we can't really use the return data directly and form a `NewCalendarEventType` object from it, as at times it will be null, thus causing JSON errors. Instead, we build it by utilizing the data we already have direct access to.